runtimes/eden/src/migrations.rs

```diff
@@ -0,0 +1,77 @@
+use crate::Runtime;
+use frame_support::{migration, parameter_types, traits::OnRuntimeUpgrade, weights::Weight, BoundedVec};
+use primitives::AccountId;
+
+#[cfg(feature = "try-runtime")]
+use frame_support::traits::OnRuntimeUpgradeHelpersExt;
+
+// MaxMembers is chosen based on what used to be the MaxMembers param for the pallet ValidaorsSet
+// We have intentionally used the same number for MaxInvulnerables for the pallet CollatorSelection
+// So the BoundedVec storage in these two pallets remain compatible.
+parameter_types! {
+	pub const MaxMembers: u32 = 50;
+}
+
+pub struct MoveValidatorsSetToInvulnerables;
+impl OnRuntimeUpgrade for MoveValidatorsSetToInvulnerables {
+	fn on_runtime_upgrade() -> Weight {
+		if let Some(validators) =
+			migration::take_storage_value::<BoundedVec<AccountId, MaxMembers>>(b"ValidatorsSet", b"Members", &[])
+		{
+			log::info!(target: "try-runtime", "ValidatorsSet::Members are {:?}", validators);
+
+			migration::put_storage_value(b"CollatorSelection", b"Invulnerables", &[], validators);
+
+			let clear_prime_result = migration::clear_storage_prefix(b"ValidatorsSet", b"Prime", &[], None, None);
+			if clear_prime_result.maybe_cursor.is_none() {
+				log::info!(target: "try-runtime", "ValidatorsSet::Prime with {} unique entries is removed", clear_prime_result.unique);
+			} else {
+				log::error!(target: "try-runtime", "Failed to remove ValidatorsSet::Prime completely");
+			}
+
+			let clear_poa_result = migration::clear_storage_prefix(b"Poa", b"StorageVersion", &[], None, None);
+			if clear_poa_result.maybe_cursor.is_none() {
+				log::info!(target: "try-runtime", "Poa::StorageVersion with {} unique entries is removed", clear_poa_result.unique);
+			} else {
+				log::error!(target: "try-runtime", "Failed to remove Poa::StorageVersion completely");
+			}
```

Comment:
I think you must have a way to clear all the storage values under a certain prefix to make this a little less verbose

Reply:
I couldn't find one in those migration helper functions. They all need both the module and the item to be specified. I tried to see if an empty item name can do the job but it didn't. Do you know how we can do that?

Reply:
Check this one https://paritytech.github.io/substrate/master/frame_support/storage/migration/fn.clear_storage_prefix.html

Reply:
I'm using the exact same function, but there you need to specify the "item". I don't know how we can remove the while storage a pallet in a single call.